pkg/kubelego/watch.go

```diff
@@ -27,13 +27,25 @@ func ingressWatchFunc(c *kubernetes.Clientset, ns string) func(options k8sMeta.L
 	}
 }
 
-func (kl *KubeLego) requestReconfigure() {
-	kl.workQueue.Add(true)
+// requestReconfigure will trigger a resync of *all* ingress resources.
+func (kl *KubeLego) requestReconfigure() error {
+	allIng, err := ingress.All(kl)
+	if err != nil {
+		return err
+	}
+	for _, ing := range allIng {
+		key, err := cache.MetaNamespaceKeyFunc(ing.Object)
+		if err != nil {
+			return err
+		}
+		kl.workQueue.AddRateLimited(key)
+	}
+	return nil
```

Comment:
This function is only used by the renewal ticker in order to trigger a resync of ingress resources.

Previously, it resynced all resources at once, with no kind of rate limit applied.

This changes that to add it via the rate limited queue. For items that are already failing, this will increment the rate limit slightly, however this is probably not an issue.

For the items the are currently not 'known' by the rate limiter (i.e. succeeding ingresses), they will be checked one by one after a delay of 10 minutes, which is also acceptable imo.